app/scripts/modules/core/src/search/externalSearch.registry.ts

```diff
@@ -30,22 +34,47 @@ export interface IExternalSearchConfig {
 }
 
 export class ExternalSearchRegistry {
-  private registry: {[key: string]: IExternalSearchConfig} = {};
+  private registry: IExternalSearchConfig[] = [];
+
+  public getAll(): IExternalSearchConfig[] {
+    return this.registry;
+  }
 
   public register(searchConfig: IExternalSearchConfig) {
-    const type = searchConfig.searchResultType;
+    const type = searchConfig.type;
     searchResultTypeRegistry.register(type);
     urlBuilderRegistry.register(type.id, searchConfig.urlBuilder);
-    this.registry[type.id] = searchConfig;
+    this.registry.push(searchConfig);
```

Comment:
what happens if there are two search configs with the same type/id?

Reply:
I bet the first one in would win instead of the last one in

Reply:
I bet that's fine